src/Resources/config/block.php

```diff
@@ -27,34 +27,42 @@
         ->tag('sonata.block')
         ->args([
             new ReferenceConfigurator('twig'),
-            new ReferenceConfigurator('service_container'),
+            new ReferenceConfigurator('sonata.media.pool'),
+            new ReferenceConfigurator('sonata.media.admin.media'),
             new ReferenceConfigurator('sonata.media.manager.media'),
+            new ReferenceConfigurator('sonata.admin.field_description_factory.orm'),
         ]);
 
     $services->set('sonata.media.block.feature_media', FeatureMediaBlockService::class)
         ->public()
         ->tag('sonata.block')
         ->args([
             new ReferenceConfigurator('twig'),
-            new ReferenceConfigurator('service_container'),
+            new ReferenceConfigurator('sonata.media.pool'),
+            new ReferenceConfigurator('sonata.media.admin.media'),
             new ReferenceConfigurator('sonata.media.manager.media'),
+            new ReferenceConfigurator('sonata.admin.field_description_factory.orm'),
```

Comment:
Ideally I would love to not couple with orm here, how should I do it? if Odm is installed I should use the field_description_factory from odm.

One idea could be to add an alias here to point to the correct field_description_factory
Other idea could be to name them equally on the persistence bundle (not sure if that is a good idea at all).

Maybe there are other ideas, wdyt @VincentLanglet ?

Reply:
I don't know a lot with config. Maybe @franmomu does.
I don't think using the same name is the best cause you could use both persistence bundle.
I think alias is made for this. The same way you inject an interface and not a class implementing the interface in the construct.

Reply:
This is not a problem now.